man/sd_bus_message_read.xml

```diff
@@ -0,0 +1,224 @@
+<?xml version='1.0'?>
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
+"http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd">
+
+<!--
+  SPDX-License-Identifier: LGPL-2.1+
+-->
+
+<refentry id="sd_bus_message_read"
+          xmlns:xi="http://www.w3.org/2001/XInclude">
+
+  <refentryinfo>
+    <title>sd_bus_message_read</title>
+    <productname>systemd</productname>
+  </refentryinfo>
+
+  <refmeta>
+    <refentrytitle>sd_bus_message_read</refentrytitle>
+    <manvolnum>3</manvolnum>
+  </refmeta>
+
+  <refnamediv>
+    <refname>sd_bus_message_read</refname>
+    <refname>sd_bus_message_readv</refname>
+
+    <refpurpose>Read a sequence of values from a message</refpurpose>
+  </refnamediv>
+
+  <refsynopsisdiv>
+    <funcsynopsis>
+      <funcsynopsisinfo>#include &lt;systemd/sd-bus.h&gt;</funcsynopsisinfo>
+
+      <funcprototype>
+        <funcdef>int <function>sd_bus_message_read</function></funcdef>
+        <paramdef>sd_bus_message *<parameter>m</parameter></paramdef>
+        <paramdef>char char *<parameter>types</parameter></paramdef>
+        <paramdef>...</paramdef>
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>int <function>sd_bus_message_readv</function></funcdef>
+        <paramdef>sd_bus_message *<parameter>m</parameter></paramdef>
+        <paramdef>char char *<parameter>types</parameter></paramdef>
+        <paramdef>va_list <parameter>ap</parameter></paramdef>
+      </funcprototype>
+    </funcsynopsis>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para><function>sd_bus_message_read()</function> reads a sequence of fields from
+    the D-Bus message object <parameter>m</parameter> and advances the read position
+    in the message. The type string <parameter>types</parameter> describes the types
+    of items expected in the message and the field arguments that follow. The type
+    string may be <constant>NULL</constant> or empty, in which case nothing is
+    read.</para>
+
+    <para>The type string is composed of the elements described in
+    <citerefentry><refentrytitle>sd_bus_message_append</refentrytitle><manvolnum>3</manvolnum></citerefentry>,
+    i.e. basic and container types. It must contain zero or more single "complete
+    types". The type string is <constant>NUL</constant>-terminated.</para>
+
+    <para>For each type specified in the type string, one or more arguments need to
+    be specified after the <parameter>types</parameter> parameter, in the same order.
+    The arguments must be pointers of the appropriate type (a pointer to
+    <code>int8_t</code> for a <literal>y</literal> in the type string, a pointer to
+    <code>int32_t</code> for an <literal>i</literal>, ...). See the table below for a
+    complete list of allowed arguments. Each argument may also be
+    <constant>NULL</constant>, in which case the value is read and ignored.</para>
```

Comment:
hmm, i think we should mention explicitly here that for strings this expects a pointer to a pointer to a char... This is particular relevant as this is not symmetric to sd_bus_message_append() as that expects a pointer to a char for strings...

Reply:
I reworked this part to add the clarification. I also added a bit about what arguments are "input" and what are "output".

I also noticed that I actually got the type wrong in the table in sd_bus_read_basic().

PTAL.